pkg/evaluation/util_test.go

```diff
@@ -0,0 +1,312 @@
+				target *rest.Target
+				attr   string
+			}{
+				target: &rest.Target{
+					Identifier: identifier,
+				},
+				attr: "identifier",
+			},
+			want: reflect.ValueOf(identifier),
+		},
+		{
+			name: "check attributes",
+			args: struct {
+				target *rest.Target
+				attr   string
+			}{
+				target: &rest.Target{
+					Identifier: identifier,
+					Attributes: &map[string]interface{}{
+						"email": email,
+					},
+				},
+				attr: "email",
+			},
+			want: reflect.ValueOf(email),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := getAttrValue(tt.args.target, tt.args.attr); !reflect.DeepEqual(got.Interface(), tt.want.Interface()) {
+				t.Errorf("getAttrValue() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_findVariation(t *testing.T) {
+	trueVariation := rest.Variation{
+		Identifier: "true",
+		Value:      "true",
+	}
+	falseVariation := rest.Variation{
+
+		Identifier: "false",
+		Value:      "false",
+	}
+	type args struct {
+		variations []rest.Variation
+		identifier string
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    rest.Variation
+		wantErr bool
+	}{
+		{
+			name: "true variation",
+			args: struct {
+				variations []rest.Variation
+				identifier string
+			}{
+				variations: []rest.Variation{trueVariation, falseVariation},
+				identifier: "true",
+			},
+			want:    trueVariation,
+			wantErr: false,
+		},
+		{
+			name: "not found variation",
+			args: struct {
+				variations []rest.Variation
+				identifier string
+			}{
+				variations: []rest.Variation{trueVariation, falseVariation},
+				identifier: "hundred",
+			},
+			want:    rest.Variation{},
+			wantErr: true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := findVariation(tt.args.variations, tt.args.identifier)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("findVariation() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
+				t.Errorf("findVariation() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_getNormalizedNumber(t *testing.T) {
+	type args struct {
+		identifier string
+		bucketBy   string
+	}
+	tests := []struct {
+		name string
+		args args
+		want int
+	}{
+		{
+			name: "happy path",
+			args: struct {
+				identifier string
+				bucketBy   string
+			}{
+				identifier: "enver.bisevac@harness",
+				bucketBy:   "email",
+			},
+			want: 61,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := getNormalizedNumber(tt.args.identifier, tt.args.bucketBy); got != tt.want {
+				t.Errorf("getNormalizedNumber() = %v, want %v", got, tt.want)
+			}
+		})
+	}
+}
+
+func Test_isEnabled(t *testing.T) {
+	type args struct {
+		target     *rest.Target
+		bucketBy   string
+		percentage int
+	}
+	tests := []struct {
+		name string
+		args args
+		want bool
+	}{
+		{
+			name: "rollout from 0",
+			args: struct {
```

Comment:
looks like your re-declaring a struct here, could you use the args struct from above

Reply:
make sense